scripts/packages/VSCodeServer/src/progress.jl

```diff
@@ -1,10 +1,18 @@
 struct VSCodeLogger <: Logging.AbstractLogger end
 
+const logger_lock = ReentrantLock()
 function Logging.handle_message(j::VSCodeLogger, level, message, _module,
     group, id, file, line; kwargs...)
     isprogress = try_process_progress(level, message, _module, group, id, file, line; kwargs...) do progress
-        JSONRPC.send_notification(conn_endpoint[], "repl/updateProgress", progress)
-        JSONRPC.flush(conn_endpoint[])
+        lock(logger_lock)
+        try
+            JSONRPC.send_notification(conn_endpoint[], "repl/updateProgress", progress)
+            JSONRPC.flush(conn_endpoint[])
+        catch err
+            # it's fine to drop this
```

Comment:
Hm, why would we want to drop errors here? That seems quite risky?

Reply:
Don't think so, but I've removed the catch block.